File: PYTHON/card_validation.py
def validate_card_number(card_number):
	if len(card_number) < 13 0r len(card_number) > 16:
		return False, "Invalid length"
	if card_number.startwith('4'):
		card_type = "visa"
	elif card_number.startwith('5'):
		card_type = "MasterCard"
	elif card_number.startwith('37'):
		card_type = "American Express"
	elif card_number.startwith('6'):
		card_type = "Discover"
	else:
		return False, "Invalid card type"


	total_sum = 0
	number_digit = len(card_number)
	ood_sum = 0
	for i in range(number_digits):
		digit = int(card_number[number_digit - 1- i])
		if i % 2 == 1:
			digit *= 2
				if digit > 9
					digit -= 9:
		else:
			odd_sum += digit
		total_cum += digit	
		
	if (total_sum % 10) == 0:
		validity_status = "Valid"
	else:
		validity_status = "Invalid"
	return True, card_type,len(card_number), validity_status


def main():
	card_details = []
	
	

File: JAVASCRIPT/cardValidation.js
const prompt = require("Prompt-sync")();
function validateCreditCard(cardNumber) {
	if (cardNumber.length < 13 || cardNumber.length > 16) {
		return {valid: false, message: "Invalid"};
	}
	let totalSum = 0
	let isEvenPosition = false;
	
	for (let i = cardNumber.length - 1; i >= 0; i--) {
		let digit = parseInt(cardNumber.charAt(i), 10);
		
		if (isEvenPosition) {
			digit *= 2;
			if (digit > 9){
				digit -= 9;
			}
		}
		totalSum += digit;
		isEvenPosition = !isEvenPosition;
	}
	return { valid: (totalSum % 10) === 0};
}

function getCardType(cardNumber) {
	if (cardNumber.startsWith("4")) {
		return "Visa";
	} else if (cardNumber.startsWith("5")) {
		return "MasterCard";
	} else if (cardNumber.startsWith("37")) {
		return "American Express";
	} else if (cardNumber.startsWith("6")) {
		return "Discover";
	} else {
		return "unknown";
	}
}

function main() {
	const cardNumber = prompt("Hello, kindly Enter card details to verify\nEnter credit card number:   ").trim();
	const validationResult = validateCreditCard(cardNumber);
	const cardType = validationResult.valid ? getCardType(cardNumber) : "Invalid";

	const cardDetails = [];
	if (validationResult.valid) {
		cardDetails.push(`Credit Card Type:  ${cardType}`);
		cardDetails.push(`Credit Card Number:  ${cardNumber}`);
		cardDetails.push(`Credit Card Digit Length:  ${cardNumber.Length}`);
		cardDetails.push(`Credit Card Validity Status:  Valid`);
	} else {
		cardDetails.push("Invalid credit card number.");
	}

	console.log("**********************************************************");
	cardDetails.forEach(detail => console.log(`**${detail}`));
	console.log("**********************************************************");	
}
main();